src/Page.js

```diff
@@ -516,9 +518,13 @@ Page.prototype.collectFrontMatter = function (includedPage) {
   } else {
     // Page is addressable but no front matter specified
     this.frontMatter = {
-      src: this.src,
-      title: this.title || '',
-      layout: LAYOUT_DEFAULT_NAME,
+      ...{
```

Comment:
Is there a reason why this is in an object? 
Perhaps we could do

```js
const defaultAttributes = {
        src: this.src,
        title: this.title || '',
        layout: LAYOUT_DEFAULT_NAME,
}

this.frontMatter = {
    ...defaultAttributes 
```

Additionally, this could collapse with the logic above? the title check and layout check seems redundant there.

Reply:
On second thought perhaps we could do something like
```js
let front matter = { default }
if (specified front matter exists)
    front matter = specified front matter

// handle overrides
```

Reply:
Hmm this is a bit tricky because the default src and title actually takes priority over the specified front matter, we only use the specified one if they are undefined.

Overriding with ... seems to include undefineds:

```
a = { x: 5 }
b = { x: undefined }
{...a, ...b } // { x: undefined }
```